libraries/abstractions/secure_sockets/test/iot_test_tcp.c

```diff
@@ -2961,6 +2962,61 @@ TEST( Full_TCP, AFQP_SOCKETS_htons_HappyCase )
 }
 /*-----------------------------------------------------------*/
 
+TEST( Full_TCP, test_dns_cache_multiple_addresses )
+{
+    BaseType_t xResult = pdFAIL;
+    uint32_t i;
+    uint32_t j;
+    uint32_t ulIPAddress;
+    uint32_t ulUnique;
+    uint32_t ulNumUniqueIPAddresses = 0;
+
+    /* Resolve the AWS IoT broker endpoint, which will have multiple IP addresses */
+
+    uint32_t ulIPAddresses[ 10 ] = { 0UL };
+
+    tcptestPRINTF( ( "Starting %s.\r\n", __FUNCTION__ ) );
+    /*
+     * Resolve the broker endpoint to an array of IP addresses. Each
+     * subsequent call will return one of the addresses which the
+     * name resolves to.  Call once more than the maximum number of addresses
+     * per cache entry to ensure that wraparound works.
+     *
+     * NOTE: Resolving all addresses associated with the AWS IoT broker endpoint
+     * can take some time, so allow 60 seconds to collect all of them.
+     */
+    for( i = 0 ; i < 60  ; i ++ )
+    {
+        ulIPAddress = SOCKETS_GetHostByName( clientcredentialMQTT_BROKER_ENDPOINT );
+
+        for( j = 0, ulUnique = 1 ; j < ulNumUniqueIPAddresses ; j++ )
+        {
+            if( ulIPAddresses[ j ] == ulIPAddress )
+            {
+                ulUnique = 0;
+            }
+        }
+        if( ( ulUnique == 1 ) && ( ulNumUniqueIPAddresses < 10 ) )
```

Comment:
Why are we trying to obtain 10 different addresses while getting only 4 is required to pass the test?

Reply:
Good point, those numbers should match – will change.